x-pack/plugins/security_solution/public/common/components/event_details/endpoint_response_actions_tab.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiNotificationBadge, EuiSpacer } from '@elastic/eui';
+import { ActionsLogTable } from '../../../management/components/endpoint_response_actions_list/components/actions_log_table';
+import { useGetEndpointActionList } from '../../../management/hooks';
+import type { ExpandedEventFieldsObject, RawEventData } from './types';
+import { EventsViewType } from './event_details';
+import * as i18n from './translations';
+
+import { expandDottedObject } from '../../../../common/utils/expand_dotted';
+import { useIsExperimentalFeatureEnabled } from '../../hooks/use_experimental_features';
+import { RESPONSE_ACTION_TYPES } from '../../../../common/detection_engine/rule_response_actions/schemas/response_actions';
+
+const TabContentWrapper = styled.div`
+  height: 100%;
+  position: relative;
+`;
+
+export const useEndpointResponseActionsTab = ({
+  rawEventData,
+}: {
+  rawEventData?: RawEventData;
+}) => {
+  const responseActionsEnabled = useIsExperimentalFeatureEnabled('endpointResponseActionsEnabled');
+
+  const {
+    data: actionList,
+    isFetching,
+    isFetched,
+  } = useGetEndpointActionList(
+    {
+      alertIds: [rawEventData?._id ?? ''],
+      withRuleActions: true,
+    },
+    { retry: false, enabled: rawEventData && responseActionsEnabled }
+  );
+
+  const totalItemCount = useMemo(() => actionList?.total ?? 0, [actionList]);
+
+  if (!rawEventData || !responseActionsEnabled) {
+    return;
+  }
+
+  const expandedEventFieldsObject = expandDottedObject(
+    rawEventData.fields
+  ) as ExpandedEventFieldsObject;
+
+  const responseActions =
+    expandedEventFieldsObject?.kibana?.alert?.rule?.parameters?.[0].response_actions;
```

Comment:
Will the parameters array always have at least 1 value or would it be safer to be `[0]?.response_actions`?

Reply:
This is a very good question, thanks! I wasn't sure so took me a while to check:
-  rule always contains paramaters []. Both with rawEventData and expandedEventFieldsObject. I think we're safe with the current state 👍  

![Zrzut ekranu 2023-03-27 o 14 40 13](https://user-images.githubusercontent.com/16632552/227957940-4b6c80b0-cb84-4906-86f8-932cc157aba8.png)


![Zrzut ekranu 2023-03-27 o 14 45 07](https://user-images.githubusercontent.com/16632552/227957932-c348fdf6-d000-452f-b03e-b893084ce5fd.png)